### PR TITLE
Automatically exec js commands

### PR DESCRIPTION
**Pull request by gagdiez**
This PR tries to automatically execute the CLI commands that are suggested after somebody tries to run a JS commands.

I will put is as a draft for now, since my initially proposed code might be very wrong.

If the core devs feel it would be faster for them to code this instead of reviewing my code, please feel free to close this PR.

@race-of-sloths

### CODE REVIEW

File: src/main.rs
Comment (gagdiez) [inline src/main.rs:117]: used `unwrap` here because it should always work

File: src/main.rs
Comment (gagdiez) [inline src/main.rs:134]: do not really understand what was happening here, but removing it does not seam to impact the tests or functionallity

Comment (frol) [reply src/main.rs:134]: @gagdiez Please, bring it back. Without these, if you just enter a slightly invalid JS command you will get an unhelpful error message.

File: src/main.rs
Comment (frol) [reply src/main.rs:117]: You can replace `try_parse_from + unwrap` with just `parse_from`, if I remember correctly.

Comment (gagdiez) [reply src/main.rs]: apparently, `Cmd::parse_from`  does not exist, so I used `Parser::parse_from`

File: src/main.rs
Comment (gagdiez) [reply src/main.rs:134]: I understood what was happening. I implemented a simpler logic that seams to handle errors correctly.

the logic now is:

```
if RS and JS cmd failed to parse the subcommand, we try an external sub-command execution

if JS successfully managed to parse the subcommand, then we let the `JSCmd` handle the error

otherwise we let `RS Cmd` handle the error
```

since there only 3 possibilities (both failed to parse the sub-command, JS failed, RS failed), then we have covered all cases.

I tested it manually and it seams to work fine

**Comment by gagdiez** [comment]
@frol this is ready for a second review

**Review by frol — COMMENTED** [review]
@gagdiez I noticed a minor bug with `call` (and probably `view`) command:

image

In JS CLI it was totally fine to skip arguments. Let's use `text-args` instead of `json-args` OR pass `{}` instead of an empty string to `json-args`.

File: src/js_command_match/mod.rs
Comment (gagdiez) [reply src/js_command_match/mod.rs:67]: good point, I put them back with the deprecation warning

**Comment by gagdiez** [comment]
@frol fixed the `view` and `call` by always using `json-args` and having as default `{}`

File: src/js_command_match/mod.rs
Comment (frol) [inline src/js_command_match/mod.rs:35]: The `stake` command has `validator-stake` in JS version: https://github.com/near/near-cli/blob/31567276536c35f00c4a1a08a362b51cdf7fbb27/commands/validators/stake.js#L10-L11

Let's add this alias back to Stack command

File: src/js_command_match/mod.rs
Comment (frol) [inline src/js_command_match/mod.rs:36]:
```suggestion
    #[clap(alias("validator-stake"))]
    Stake(self::deprecated::StakeArgs),
```

**Review by frol — APPROVED** [review]
@race-of-sloths score 8

**Comment by race-of-sloths** [comment]
@gagdiez Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/gagdiez?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/gagdiez?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/gagdiez)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/gagdiez)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 8!
You have received 80 Sloth points for this contribution

**Pull request merged**
